Integrations/EDL/EDL.py

```diff
@@ -167,20 +207,121 @@ def find_indicators_to_limit_loop(indicator_query: str, limit: int, total_fetche
     return iocs, next_page
 
 
-def create_values_out_dict(iocs: list) -> dict:
+def ip_groups_to_ranges(ip_range_groups: list, collapse_ips: str):
+    """Collapse ip groups list to ranges or CIDRs
+
+    Args:
+        ip_range_groups (list): a list of lists containing connected IPs
+        collapse_ips (str):  Whether to collapse to Ranges or CIDRs.
+
+    Returns:
+        list. a list to Ranges or CIDRs.
+    """
+    ip_ranges = []  # type:List
+    for group in ip_range_groups:
+        # handle single ips
+        if len(group) == 1:
+            ip_ranges.append(str(group[0]))
+            continue
+
+        min_ip = group[0]
+        max_ip = group[-1]
+        if collapse_ips == COLLAPSE_TO_RANGES:
+            ip_ranges.append(str(min_ip) + "-" + str(max_ip))
+
+        elif collapse_ips == COLLAPSE_TO_CIDR:
+            moved_ip = False
+            # CIDR must begin with an even LSB
+            # if the first ip does not - separate it from the rest of the range
+            if (int(str(min_ip).split('.')[-1]) % 2) != 0:
+                ip_ranges.append(str(min_ip))
+                min_ip = group[1]
+                moved_ip = True
+
+            # CIDR must end with uneven LSB
+            # if the last ip does not - separate it from the rest of the range
+            if (int(str(max_ip).split('.')[-1]) % 2) == 0:
+                ip_ranges.append(str(max_ip))
+                max_ip = group[-2]
+                moved_ip = True
+
+            # if both min and max ips were shifted and there are only 2 ips in the range
+            # we added both ips by the shift and now we move to the next  range
+            if moved_ip and len(group) == 2:
+                continue
+
+            else:
+                ip_ranges.append(str(iprange_to_cidrs(min_ip, max_ip)[0].cidr))
+
+    return ip_ranges
+
+
+def ips_to_ranges(ips: list, collapse_ips):
+    """Collapse IPs to Ranges or CIDRs.
+
+    Args:
+        ips (list): a list of IP strings.
+        collapse_ips (str): Whether to collapse to Ranges or CIDRs.
+
+    Returns:
+        list. a list to Ranges or CIDRs.
+    """
+    ips_range_groups = []  # type:List
+    ips = sorted(ips)
+
+    if len(ips) > 0:
+        ips_range_groups.append([ips[0]])
+
+    if len(ips) > 1:
+        for ip in ips[1:]:
+            appended = False
+
+            for group in ips_range_groups:
+                if IPAddress(int(ip) + 1) in group or IPAddress(int(ip) - 1) in group:
+                    group.append(ip)
+                    sorted(group)
+                    appended = True
+
+            if not appended:
+                ips_range_groups.append([ip])
+
+    return ip_groups_to_ranges(ips_range_groups, collapse_ips)
```

Comment:
Consider splitting this function into three cases: ip_groups_to_ranges, ip_groups_to_cidr or just returning list as is

Reply:
if they are not collapsed they wont get here - I'll separate ranges from cidrs